corehq/form_processor/models.py

```diff
@@ -722,6 +722,23 @@ def get_obj_by_id(cls, case_id):
         from corehq.form_processor.backends.sql.dbaccessors import CaseAccessorSQL
         return CaseAccessorSQL.get_case(case_id)
 
+    @property
+    @memoized
+    def parent(self):
+        """
+        Returns the parent case if one exists, else None.
+        NOTE: This property should only return the first parent in the list
+        of indices. If for some reason your use case creates more than one,
+        please write/use a different property.
```

Comment:
I don't think it's necessary to check the relationship field since but also couldn't hurt.

If we needed this to work with extension cases as well then we'd have to expand the notion of 'parent' to include 'host' cases in which case the conditions would be:
* relationship='child', identifier='parent' (normal parent / child)
* relationship='extension' (host / extension)